gloo/docs/enterprise/authentication/jwt.md

```diff
@@ -0,0 +1,258 @@
+---
+title: JWT and Access Control
+weight: 2
+description: JWT verification and Access Control (without an external auth server)
+---
+
+## Pre-requesites
+
+We will use the following utilities
+
+- minikube
+- jq
+- gloo enterprise version v0.13.16 or later.
+
+## Initial setup
+
+Start minikube:
+```
+minikube start
+```
+
+Install gloo-enterprise and create a virtual service and an example app:
+```shell
+glooctl install gateway --license-key <YOUR KEY>
+```
+
+Wait for the deployment to finish:
+```shell
+kubectl -n gloo-system rollout status deployment/discovery
+kubectl -n gloo-system rollout status deployment/gateway
+kubectl -n gloo-system rollout status deployment/gloo
+kubectl -n gloo-system rollout status deployment/gateway-proxy
+```
+
+Install the petstore demo app and add a route and test that everything so far works (you may need to wait a minutes until all the gloo containers are initialized):
+```shell
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo/master/example/petstore/petstore.yaml
+glooctl add route --name default --namespace gloo-system --path-prefix / --dest-name default-petstore-8080 --dest-namespace gloo-system
+URL=$(glooctl proxy url)
+```
+
+Test that everything so far works:
+```
+curl $URL/api/pets/
+[{"id":1,"name":"Dog","status":"available"},{"id":2,"name":"Cat","status":"pending"}]
+```
+
+## Setting up JWT authorization
+
+Let's create a test pod, with a different service account. We will use this pod in the guide to test
+access with the new service account credentials.
+
+```shell
+kubectl create serviceaccount svc-a
+kubectl run --generator=run-pod/v1 test-pod --image=fedora:30 --serviceaccount=svc-a --command sleep 10h
+```
+
+### Anatomy of kuberentes service account
+
+When kuberentes starts a pod, it automatically attaches to it a JWT (JSON Web Token), that allows 
+for authentication with the credentials of the pod's service account.
+Inside the JWT are *claims* that provide idtentity information, and a signature to verification.
+
+To verify these JWT, the kubernetes api server is provided with a public key. We can use this public 
+key to perform JWT verification for kubernetes service accounts in Gloo.
+
+Let's see the claims for `svc-a` - the service account we just created:
+
+```shell
+kubectl exec test-pod cat /var/run/secrets/kubernetes.io/serviceaccount/token | cut -d. -f2 | base64 -d 2>/dev/null|jq .
```

Comment:
```suggestion
kubectl exec test-pod cat /var/run/secrets/kubernetes.io/serviceaccount/token | cut -d. -f2 | base64 --decode 2>/dev/null|jq .
```

awesome command!
`-d` does not work on mac but `--decode` does

Reply:
thanks!